source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -257,6 +291,44 @@ export abstract class CardElement {
         // Does nothing in base implementation
     }
 
+    handleOverflow(maxHeight: number) {
+        if (this.isVisible || this.isHiddenDueToOverflow()) {
+            var handled = this.truncateOverflows(maxHeight);
+
+            // Even if we were unable to truncate the element to fit this time,
+            // it still could have been previously truncated
+            this._truncatedDueToOverflow
+                = handled || this._truncatedDueToOverflow;
+
+            if (!handled) {
+                this.hideElementDueToOverflow();
+            }
+            else if (handled && !this._isVisible) {
+                this.showElementHiddenDueToOverflow();
+            }
+        }
+    }
+
+    resetOverflow(): boolean {
+        var sizeChanged = false;
+
+        if (this._truncatedDueToOverflow) {
+            this.undoOverflowTruncation();
+            this._truncatedDueToOverflow = false;
+            sizeChanged = true;
+        }
+
+        if (this.isHiddenDueToOverflow) {
+            this.showElementHiddenDueToOverflow();
+        }
+
+        return sizeChanged;
+    }
+
+    isOnPage(): boolean {
+        return !!(this._renderedElement && this._renderedElement.offsetHeight);
```

Comment:
Is the "!!" a typo? If not, I am not familiar with what it means, if you don't mind educating me :)

Reply:
It coerces `this._renderedElement && this._renderedElement.offsetHeight` into a boolean -- otherwise the result of evaluating that expression would be either false or a number.

Reply:
So is it equivalent to ```this._renderedElement && this._renderedElement.offsetHeight > 0```? If so, can we use that syntax instead? I find the these kinds of syntax shortcuts difficult to read.
Thanks for the explanation, I leanred something new today.

Reply:
Yep, that would work as well. Will update.